hles/extraction/src/main/scala/org/broadinstitute/monster/dap/CslbExtractionPipeline.scala

```diff
@@ -15,9 +15,18 @@ object CslbExtractionPipeline extends ScioApp[Args] {
   // Magic marker for "completed".
   // NB: We are purposefully excluding the recruitment_fields_complete -> 2
   // mapping, as that conflicts with the CSLB data
-  val extractionFilters: List[FilterDirective] = List(
-    FilterDirective("canine_social_and_learned_behavior_complete", FilterOps.==, "2")
-  )
+  def extractionFiltersGenerator(args: Args): List[FilterDirective] =
+    List(
+      FilterDirective("canine_social_and_learned_behavior_complete", FilterOps.==, "2")
+    ) ++
+      args.startTime
+        .map(start =>
+          List(FilterDirective("cslb_date", FilterOps.>, RedCapClient.redcapFormatDate(start)))
+        )
+        .getOrElse(List()) ++
+      args.endTime
+        .map(end => List(FilterDirective("cslb_date", FilterOps.<, RedCapClient.redcapFormatDate(end))))
+        .getOrElse(List())
```

Comment:
i am positive that there's a simpler way to conditionally add a value to a list inline, but i don't know enough scala to know it

Reply:
I am not sure myself; a case match might be slightly more idiomatic?

```suggestion
  List(
      FilterDirective("canine_social_and_learned_behavior_complete", FilterOps.==, "2")
    ) ++
      (args.startTime match {
        case Some(start) =>
          List(FilterDirective("cslb_date", FilterOps.>, RedCapClient.redcapFormatDate(start)))
        case _ => List()
      }) ++
      (args.endTime match {
        case Some(end) =>
          List(FilterDirective("cslb_date", FilterOps.<, RedCapClient.redcapFormatDate(end)))
        case _ => List()
      })
```

Reply:
if memory serves, scala best practices discourage using `match` on `Option` types in favor of the `map/getOrElse` pattern i'm using here